hub/auto/unstructured/image_classification.py

```diff
@@ -0,0 +1,145 @@
+import warnings
+import numpy as np
+from pathlib import Path
+import os
+import glob
+from typing import Dict, List, Sequence, Tuple, Union
+
+from hub.util.exceptions import InvalidPathException, TensorInvalidSampleShapeError
+from hub import Dataset
+
+from tqdm import tqdm  # type: ignore
+
+from .base import UnstructuredDataset
+
+import hub
+
+IMAGES_TENSOR_NAME = "images"
+LABELS_TENSOR_NAME = "labels"
+
+
+def _get_file_paths(directory: Path, relative_to: Union[str, Path] = "") -> List[Path]:
+    if os.path.isdir(directory):
+        g = glob.glob(os.path.join(directory, "**"), recursive=True)
+        file_paths = []
+        for path_str in g:
+            if os.path.isfile(path_str):
+                path = Path(path_str)
+                if relative_to:
+                    relative_path = Path(path).relative_to(directory)
+                else:
+                    relative_path = path
+                file_paths.append(relative_path)
+        return file_paths
+    else:
+        raise InvalidPathException(directory)
+
+
+def _class_name_from_path(path: Path) -> str:
+    return path.parts[-2]
+
+
+def _set_name_from_path(path: Path) -> str:
+    return path.parts[-3]
+
+
+class ImageClassification(UnstructuredDataset):
+    def __init__(self, source: str):
+        """Convert an unstructured dataset to a structured dataset.
+        Args:
+            source (str): The full path to the dataset.
```

Comment:
```suggestion
        """Convert an unstructured dataset to a structured dataset.
        
        Args:
            source (str): The full path to the dataset.
```

make sure you add an empty line above all sections in docstrings. otherwise, our darglint doesn't properly register them.